flowclient/flowclient/client.py

```diff
@@ -455,6 +455,320 @@ def daily_location(
+    },
+    tower_hour_of_day_scores: List[float] = [
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        0,
+        0,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        0,
+        0,
+        0,
+        0,
+        -1,
+        -1,
+        -1,
+    ],
+    subscriber_subset: Union[dict, None] = None,
+):
+    if subscriber_subset is None:
+        subscriber_subset = "all"
+    return {
+        "query_kind": "meaningful_locations",
+        "params": {
+            "label": label,
+            "clusters": {
+                "query_kind": "hartigan_cluster",
+                "params": {
+                    "radius": tower_cluster_radius,
+                    "buffer": tower_cluster_buffer,
+                    "call_threshold": tower_cluster_call_threshold,
+                    "call_days": {
+                        "query_kind": "call_days",
+                        "params": {
+                            "subscriber_locations": {
+                                "query_kind": "subscriber_locations",
+                                "params": {
+                                    "start": start_date,
+                                    "stop": stop_date,
+                                    "level": "versioned-site",
+                                    "subscriber_subset": subscriber_subset,
+                                },
+                            }
+                        },
+                    },
+                },
+            },
+            "scores": {
+                "query_kind": "event_score",
+                "params": {
+                    "score_hour": tower_hour_of_day_scores,
+                    "score_dow": tower_dow_scores,
+                    "start": start_date,
+                    "stop": stop_date,
+                    "level": "versioned-site",
+                    "subscriber_subset": subscriber_subset,
+                },
+            },
+            "labels": labels,
+        },
+    }
+
+
+def meaningful_locations_aggregate(
+    start_date: str,
+    stop_date: str,
+    label: str,
+    labels: Dict[str, Dict[str, dict]],
+    aggregation_unit: str,
+    tower_cluster_radius: float = 1.0,
+    tower_cluster_buffer: float = 0.0,
+    tower_cluster_call_threshold: int = 0,
+    tower_dow_scores: Dict[str, float] = {
+        "monday": 1,
+        "tuesday": 1,
+        "wednesday": 1,
+        "thursday": 0,
+        "friday": -1,
+        "saturday": -1,
+        "sunday": -1,
+    },
+    tower_hour_of_day_scores: List[float] = [
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        -1,
+        0,
+        0,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        0,
+        0,
+        0,
+        0,
+        -1,
+        -1,
+        -1,
+    ],
+    subscriber_subset: Union[dict, None] = None,
+) -> dict:
```

Comment:
Worth adding a docstring to explain the meaning of parameters (or alternatively to refer to some other piece of documentation that explains them)? Esp. in light of the meaning of `label` vs. `labels` as mentioned above.